### ISSUE TITLE
Error checking mailto: based authorization

### ISSUE THREAD

**Issue by angelo-v** [solid/acl-check#24]
The root .acl file of NSS contains a `mailto:` based authorization like this:

```
# Optional owner email, to be used for account recovery:
acl:agent <mailto:john.doe@mail.example>;
```
When trying to  create a file in the root of the POD, it fails and the following error appears in the NSS logs:

```
could not fetch owner doc NamedNode {                                                                                                   
  termType: 'NamedNode',                                                                                                                                                        
  value: 'mailto:john.doe@mail.example' } { [HTTPError: File not found: /opt/solid/datanull]                                                                                 
  name: 'HTTPError',                                                                                                                                                            
  message: 'File not found: /opt/solid/datanull',                                                                                                                               
  status: 404 }
```
When I remove  the `mailto:` based `acl:agent` it works, but this is only a workarround.

**Comment by angelo-v** [comment]
Would be cool to have this fixed, since it is preventing people to use cool apps like solid-focus, which write directly to the root folder. Just had to give support to someone who was not aware of this bug and wondering why it did not work.

**Comment by michielbdejong** [comment]
ok, I don't remember how we triaged this 4 weeks ago because from your description it does sound like a bug. We should have added a note on the issue, sorry about that. Would you like to have a go at fixing this yourself? If not, we'll triage it again on Monday, and then either @jaxoncreed or myself will add it to our queue.

**Comment by angelo-v** [comment]
Sorry, cannot handle this myself at the moment.

**Comment by Hyuryu** [comment]
I experienced a similar problem using the nodejs 5.x server

When a mail is specified in an acl file, applications do not have any access mode.
Here is the response received
> 403: All Required Access Modes Not Granted

The problem can be bypassed by commenting the mail line in the root acl file ("acl:agent <mailto:fake@example.com>;").

The problem seems to come from acl-check.js: getTrustedModesForOrigin does not seem to return any results because it somehow fails while trying to get the modes authorized by the mail acl:agent (e. g. mailto:fake@example.com)

> solid:ACL Using ACL https://192.168.1.42:8454/.acl for ./.acl +0ms
  solid:ACL    2 direct authentications about <https://192.168.1.42:8454/> +16ms
  solid:ACL could not fetch owner doc NamedNode { termType: 'NamedNode', value: 'mailto:fake@example.com' } Only HTTP(S) protocols are supported +20ms
  solid:ACL accessDenied: checking access to <https://192.168.1.42:8454/> by <https://persocloud.solidweb.org/profile/card#me> and origin <https://hyuryu.solid.community> +477ms
  solid:ACL trustedOrigins <https://192.168.1.42:8454>, modes  +3ms
  solid:ACL    2 direct authentications about <https://192.168.1.42:8454/> +6ms
  solid:ACL     Agent or group: Ok, its public. +6ms
  solid:ACL       Mode allowed: <http://www.w3.org/ns/auth/acl#Read> +3ms
  solid:ACL    Checking auth <https://192.168.1.42:8454/.acl#owner> with agent <https://persocloud.solidweb.org/profile/card#me> +3ms
  solid:ACL     Agent explicitly authenticated. +4ms
  solid:ACL      Origin check FAILED. Origin not trusted. +3ms
  solid:ACL       Check failed: Origin Unauthorized +2ms
  solid:ACL accessDenied: modeURIorReasons: ["http://www.w3.org/ns/auth/acl#Read","Origin Unauthorized"] +1ms
  solid:ACL  checking <http://www.w3.org/ns/auth/acl#Control> +2ms
  solid:ACL   MODE REQUIRED NOT ALLOWED: <http://www.w3.org/ns/auth/acl#Control> Denying with All Required Access Modes Not Granted +1ms
  solid:ACL Read access denied to https://persocloud.solidweb.org/profile/card#me: 403 - All Required Access Modes Not Granted +5ms

**Comment by michielbdejong** [comment]
cc @jaxoncreed

**Comment by michielbdejong** [comment]
This is urgent, it's causing errors in production.

**Comment by michielbdejong** [comment]
Oh bummer, from https://github.com/solid/acl-check/commits/master it looks like this was fixed in thie module, but we didn't update NSS to use the latest version? Sorry about that!

**Comment by jaxoncreed** [comment]
fixed in #1242
